dspace-spring-rest/src/main/java/org/dspace/app/rest/utils/Utils.java

```diff
@@ -219,4 +236,60 @@ public static String getFileName(MultipartFile multipartFile)
             return multipartFile.getName();
         }
     }
+
+    private List<DSpaceObject> constructDSpaceObjectList(Context context,
+                                                         HttpServletRequest request) throws IOException {
+        List<String> list = readFromRequest(request);
+
+        List<DSpaceObject> dSpaceObjects = new LinkedList<>();
+        for (String string : list) {
+            String uuid = string.substring(string.lastIndexOf('/') + 1);
+            try {
+                for (DSpaceObjectService dSpaceObjectService : dSpaceObjectServices) {
+                    DSpaceObject dSpaceObject = dSpaceObjectService.find(context, UUIDUtils.fromString(uuid));
+                    if (dSpaceObject != null) {
+                        dSpaceObjects.add(dSpaceObject);
+                        break;
+                    }
+                }
+            } catch (SQLException e) {
+                log.error(e.getMessage(), e);
+            }
+
+        }
+        return dSpaceObjects;
+    }
+
+    private List<String> readFromRequest(HttpServletRequest request) throws IOException {
+        List<String> list = new LinkedList<>();
+        Scanner scanner = new Scanner(request.getInputStream());
+
+        try {
+
+            while (scanner.hasNextLine()) {
+
+                String line = scanner.nextLine();
+                if (org.springframework.util.StringUtils.hasText(line)) {
+                    list.add(line);
+                }
+            }
+
+        } finally {
+            scanner.close();
+        }
+        return list;
+    }
+
+
+    /**
+     * This method will retrieve a list of DSpaceObjects from the Request by reading in the Request's InputStream
+     * with a Scanner and searching the InputStream for UUIDs which will then be resolved to a DSpaceObject.
+     * These will all be added to a list and returned by this method.
+     * @param request       The request of which the InputStream will be used
+     * @return              The list of DSpaceObjects that we could find in the InputStream
+     * @throws IOException  If something goes wrong
+     */
+    public List<DSpaceObject> getdSpaceObjectsFromRequest(HttpServletRequest request) throws IOException {
```

Comment:
Because uri-lists are now becoming more and more important in our REST API, I wonder if we should have Unit Tests for this method.  I'm worried that some of the parsing (UUID parsing especially, see comments above) may need tests to verify it is working as expected.

Reply:
That sounds like a very good idea, but perhaps better once we have a few implementations in this solution because this PR only contains items

Reply:
@benbosman : I'd still encourage us to start with minimal/basic unit tests rather than waiting for "perfection".  As you'll see above, I already have concerns that the UUID parsing is very fragile.  So, having unit tests to prove it is working at a basic level would be helpful here, especially since the client using the REST API can literally send *anything* in a `text/uri-list`...so we need to ensure the parsing works as we expect.

Reply:
I would consider the UUID parsing as work in progress, which is intended to be improved when more use cases for the functionality are implemented.
The concept of how this should be implemented depends on the implementations where it will be used. If it were only items, we could enforce it to be `{dspace.url}/rest/api/core/items/75c0f7f5-5a69-40e8-aa1f-8f35b1ce5a63`, but making it a reusable solution implies it has to be more flexible to also support a collection, eperson, …

Maybe it makes most sense to create the IT and only use it for sending a list of 2 items, so when future improvements are made, we can at least verify this is not broken?